### PR TITLE
Get end points

### PR DESCRIPTION
**Pull request by RaphealOlumofe**
replied to comments on git

### CODE REVIEW

File: frontend/src/main/java/starter/project/frontend/CarController.java [hunk 1]
```diff
@@ -1,36 +1,56 @@
...
+            if (car.getId() == (carId)) {
+                return car;
+            }
+        }
+        throw new ResponseStatusException(
+                HttpStatus.NOT_FOUND, "entity not found");
+    }
+
+    @GetMapping(path = "/cars/{list}")
+    public List<Car> getBatchCars(@PathVariable List list) {
```
Comment (IDWalsh) [inline frontend/src/main/java/starter/project/frontend/CarController.java]: List should have type parameters on it. Looks like `List<String>` to ensure it actually contains the same type as the Map key of String.

File: frontend/src/main/java/starter/project/frontend/CarController.java [hunk 2]
```diff
@@ -1,36 +1,56 @@
...
 
+        List<Car> newList = new ArrayList<>();
+        for (Map.Entry<String, Car> m : map.entrySet()) {
```
Comment (IDWalsh) [inline frontend/src/main/java/starter/project/frontend/CarController.java]: This is kind of inefficient. You'd expect the map of all keys in a database to be waaaaaaaay bigger than the list of ids being asked for. So it's iterating a map (which is usually a bad idea) that is possibly orders of magnitude bigger. 

It should iterate the list and do map.get (which is O(1) time) so the whole thing becomes O(N) where N is the list size.

File: frontend/src/main/java/starter/project/frontend/CarController.java (hunk 1 above)
Comment (RaphealOlumofe) [reply frontend/src/main/java/starter/project/frontend/CarController.java]: resolved

Comment (RaphealOlumofe) [reply frontend/src/main/java/starter/project/frontend/CarController.java]: resolved

File: frontend/src/main/java/starter/project/frontend/CarController.java (hunk 2 above)
Comment (RaphealOlumofe) [reply frontend/src/main/java/starter/project/frontend/CarController.java]: resolved

File: frontend/src/main/java/starter/project/frontend/CarController.java (hunk 1 above)
Comment (IDWalsh) [reply frontend/src/main/java/starter/project/frontend/CarController.java]: `(@PathVariable List list)` is still there.

Comment (RaphealOlumofe) [reply frontend/src/main/java/starter/project/frontend/CarController.java]: hmm interesting on mine its not

**Pull request merged**